src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -160,8 +148,17 @@ private bool RemoveSession(int sessionId, out ProfilerSession session)
         {
             lock (this.sessionsLock)
             {
-                if (this.monitoredSessions.Remove(sessionId, out session))
+                //cancel running XEventStream for session.
+                CancellationTokenSource targetToken;
+                if (monitoredCancellationTokenSources.Remove(sessionId, out targetToken))
+                {
+                    targetToken.Cancel();
+                }
+                if (this.monitoredSessions.Remove(sessionId, out session) && session.IsStreaming)
```

Comment:
We should remove all viewers even if the session wasn't streaming, right? Not sure if that could ever actually happen but that's what the old logic was doing. So in this case I think you can just remove the `session.IsStreaming` check completely.